internal/app/config.go

```diff
@@ -0,0 +1,244 @@
+
+func (a *App) LoadConfiguration() error {
+	a.v.SetConfigType("yaml")
+	a.v.SetEnvPrefix(model.AppName)
+	a.v.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
+	a.v.AutomaticEnv()
+
+	fh, err := os.Open(a.Config.file)
+	if err != nil {
+		return errors.Wrap(err, ErrConfig.Error())
+	}
+
+	if err = a.v.ReadConfig(fh); err != nil {
+		return errors.Wrap(err, ErrConfig.Error()+" "+a.Config.file)
+	}
+
+	if err := a.v.Unmarshal(a.Config); err != nil {
+		return errors.Wrap(err, a.Config.file)
+	}
+
+	if err := a.envVarOverrides(); err != nil {
+		return errors.Wrap(err, ErrConfig.Error())
+	}
+
+	return nil
+}
+
+func (a *App) envVarOverrides() error {
+	if a.v.GetString("log.level") != "" {
+		a.Config.LogLevel = a.v.GetString("log.level")
+	}
+
+	if a.v.GetString("app.kind") != "" {
+		a.AppKind = model.AppKind(a.v.GetString("app.kind"))
+	}
+
+	if a.v.GetString("listen.address") != "" {
+		a.Config.ListenAddress = a.v.GetString("listen.address")
+	}
+
+	if a.v.GetString("store.kind") != "" {
+		a.Config.ListenAddress = a.v.GetString("store.kind")
+	}
+
+	if len(a.Config.ConditionDefinitions) == 0 {
+		return errors.Wrap(ErrConfig, "expected one or more condition definitions")
+	}
+
+	if a.Config.EventsBorkerKind == model.NatsEventBroker {
+		if err := a.envVarNatsOverrides(); err != nil {
+			return err
+		}
+	}
+
+	switch a.Config.StoreKind {
+	case model.ServerserviceStore:
+		if err := a.envVarServerserviceOverrides(); err != nil {
+			return err
+		}
+	default:
+		return errors.Wrap(ErrConfig, "no/unknown store kind parameter")
+	}
+
+	return nil
+}
+
+// NATs streaming configuration
+var (
+	defaultNatsConnectTimeout = 100 * time.Millisecond
+)
+
+func (a *App) envVarNatsOverrides() error {
+	if a.v.GetString("nats.url") != "" {
+		a.Config.NatsOptions.URL = a.v.GetString("nats.url")
+	}
+
+	if a.Config.NatsOptions.URL == "" {
+		return errors.New("missing parameter: nats.url")
+	}
+
+	if a.v.GetString("nats.stream.user") != "" {
+		a.Config.NatsOptions.StreamUser = a.v.GetString("nats.stream.user")
+	}
+
+	if a.v.GetString("nats.stream.pass") != "" {
+		a.Config.NatsOptions.StreamPass = a.v.GetString("nats.stream.pass")
+	}
+
+	if a.v.GetString("nats.creds.file") != "" {
+		a.Config.NatsOptions.CredsFile = a.v.GetString("nats.creds.file")
+	}
+
+	if a.v.GetString("nats.stream.name") != "" {
+		a.Config.NatsOptions.Stream.Name = a.v.GetString("nats.stream.name")
+	}
+
+	if a.Config.NatsOptions.ConnectTimeout == 0 {
+		a.Config.NatsOptions.ConnectTimeout = defaultNatsConnectTimeout
+	}
+
+	return nil
+}
+
+// Server service configuration options
+
+// nolint:gocyclo // parameter validation is cyclomatic
+func (a *App) envVarServerserviceOverrides() error {
+	if a.v.GetString("serverservice.endpoint") != "" {
+		a.Config.ServerserviceOptions.Endpoint = a.v.GetString("serverservice.endpoint")
+	}
+
+	if a.v.GetString("serverservice.facilitycode") != "" {
+		a.Config.ServerserviceOptions.Endpoint = a.v.GetString("serverservice.facilitycode")
```

Comment:
typo? `.Endpoint` = `...facilitycode`

Reply:
oh good catch, fixed in [3ab8e8b](https://github.com/metal-toolbox/conditionorc/pull/1/commits/3ab8e8b08bb24946814fe0e3fea2469792695221)